modules/gh-runner-gke/main.tf

```diff
@@ -77,31 +77,42 @@ module "runner-cluster" {
   ]
 }
 
-/*****************************************
-  IAM Bindings GKE SVC
- *****************************************/
-# allow GKE to pull images from GCR
-resource "google_project_iam_member" "gke" {
-  count   = var.service_account == "" ? 1 : 0
-  project = var.project_id
-  role    = "roles/storage.objectViewer"
-  member  = "serviceAccount:${module.runner-cluster.service_account}"
+data "google_client_config" "default" {
 }
 
-data "google_client_config" "default" {
+resource "kubernetes_namespace" "arc_systems" {
+  metadata {
+    name = var.arc_systems_namespace
+  }
+}
+
+resource "kubernetes_namespace" "arc_runners" {
+  metadata {
+    name = var.arc_runners_namespace
+  }
+
+  depends_on = [helm_release.arc]
 }
 
 /*****************************************
   K8S secrets for configuring k8s runners
  *****************************************/
-resource "kubernetes_secret" "runner-secrets" {
+resource "kubernetes_secret" "gh_app_pre_defined_secret" {
   metadata {
-    name = var.runner_k8s_config
+    name      = var.gh_app_pre_defined_secret_name
+    namespace = kubernetes_namespace.arc_runners.metadata[0].name
   }
   data = {
-    repo_url   = var.repo_url
-    gh_token   = var.gh_token
-    repo_owner = var.repo_owner
-    repo_name  = var.repo_name
+    github_app_id              = var.gh_app_id
+    github_app_installation_id = var.gh_app_installation_id
+    github_app_private_key     = var.gh_app_private_key
   }
 }
+
+resource "helm_release" "arc" {
```

Comment:
Should we allow plumbing the values yaml?
https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#installing-actions-runner-controller